client/src/app/queries/tags.ts

```diff
@@ -53,11 +53,11 @@ export const useCreateTagMutation = (
     mutationFn: createTag,
     onSuccess: (res) => {
       onSuccess(res);
-      queryClient.invalidateQueries([TagsQueryKey]);
+      queryClient.invalidateQueries([TagCategoriesQueryKey, TagsQueryKey]);
```

Comment:
I debugged through the react-query code -- this only invalidates the query that matches both keys.

Effectively, this change invalidates the `useFetchTagCategories` query now instead of the `useFetchTags` query.

Reply:
AFAICT, the only way to invalidate 2 queries is to either:
 1. call invalidate twice with exactly matching `queryKey`, or
 2. setup multiple values in the initial `queryKey` and use one of those across all queries you want to invalidate at the same time.

For example:
```tsx
export const useFetchTags = () => {
  const { data, isLoading, error, refetch } = useQuery({
    queryKey: ["allTags", TagsQueryKey],
    queryFn: getTags,
    onError: (error: AxiosError) => console.log("error, ", error),
  });
  return {
    tags: data || [],
    isFetching: isLoading,
    fetchError: error,
    refetch,
  };
};

export const useFetchTagCategories = () => {
  const { data, isLoading, error, refetch } = useQuery({
    queryKey: ["allTags", TagCategoriesQueryKey],
    queryFn: getTagCategories,
    onError: (error: AxiosError) => console.log("error, ", error),
  });
  return {
    tagCategories: data || [],
    isFetching: isLoading,
    fetchError: error,
    refetch,
  };
};

export const useCreateTagMutation = (
  onSuccess: (res: any) => void,
  onError: (err: AxiosError) => void
) => {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: createTag,
    onSuccess: (res) => {
      onSuccess(res);
      queryClient.invalidateQueries(["allTags"]);
    },
    onError: (err: AxiosError) => {
      onError(err);
      queryClient.invalidateQueries(["allTags"]);
    },
  });
};
```

The extra `"allTags"` shouldn't do anything to the query itself other than make it easier to partial match on the `invalidateQueries()` call

Reply:
Yeah, option 2 here is what I meant above.